Add reference to scorecards

varunsh-coder opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [types-of-contributions.md](/main/contributing/types-of-contributions.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

Closes #14855

[OSSF Scorecards](https://github.com/ossf/scorecard) can find issues related to `token permissions` and `actions that are not pinned`. This PR helps readers of the hardening guide to take action to address the issues. 
https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/

<!--
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->


CC @josepalafox @laurentsimon

### What's being changed:

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. If you made changes to the `content` directory, a table will populate in a comment below with the staging and live article links -->

|**Source**|	**Staging**|	**Production**|	**What Changed** |
|-----------|-------------|-----------------|-----------------------|
|[content/actions/security-guides/security-hardening-for-github-actions.md](https://github.com/github/docs/blob/691b0d2d7846b469f5a43cf84e0f85a4b14383c0/content/actions/security-guides/security-hardening-for-github-actions.md)|	[Modified](https://docs-14856--patch-2.herokuapp.com/actions/security-guides/security-hardening-for-github-actions)|	[Original](https://docs.github.com/actions/security-guides/security-hardening-for-github-actions)|Added new section for OSSF Scorecards based on this [blog](https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/)


### Check off the following:

- [X] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

varunsh-coder committed:
Add reference to scorecards

scorecards can find issues related to token permissions and actions not pinned.

ramyaparimi commented:
@varunsh-coder 
Thanks so much for opening a PR! I'll get this triaged for review :zap: I edited your PR template slightly to properly link the issue and this PR 💖

varunsh-coder committed:
Merge branch 'github:main' into patch-2

varunsh-coder committed:
Update security-hardening-for-github-actions.md

Add a new section for scorecards

varunsh-coder committed:
Revert earlier changes

First commit had included scorecards within token permissions. New commit adds it as a new section. So reverting changes from first commit.

varunsh-coder commented:
I have added a new section for OpenSSF Scorecard

ethomson commented on content/actions/security-guides/security-hardening-for-github-actions.md:
```diff
@@ -202,6 +202,10 @@ The same principles described above for using third-party actions also apply to
 
+## Using OpenSSF Scorecards to secure workflows
+
+Scorecards is an automated security tool that flags risky supply chain practices. You can use the Scorecards [GitHub Action](https://github.com/marketplace/actions/ossf-scorecard-action) and [starter workflow](https://github.com/actions/starter-workflows) to follow best security practices. Once configured, the Scorecards Action runs automatically on repository changes, and alerts developers about ...
```
```suggestion
Scorecards is an automated security tool that flags risky supply chain practices. You can use the [Scorecards action](https://github.com/marketplace/actions/ossf-scorecard-action) and [starter workflow](https://github.com/actions/starter-workflows) to follow best security practices. Once configured, the Scorecards action runs automatically on repository changes, and alerts developers about risky supply chain practices using the built-in code scanning experience. The Scorecards project makes a number of checks, including script injection attacks, token permissions, and pinned actions.
```

We use little-a "actions" to refer to the composable units and big-A "Actions" to refer to the product.

varunsh-coder committed:
Update content/actions/security-guides/security-hardening-for-github-actions.md

Thanks for the suggestion...

Co-authored-by: Edward Thomson <ethomson@github.com>

ethomson reviewed (dismissed):
Thanks for the PR, @varunsh-coder !

varunsh-coder committed:
Update security-hardening-for-github-actions.md

Added a link to scorecard repo, and modified "makes a number of checks" to "runs a number of checks"

varunsh-coder commented:
@ethomson Apologies for making more changes are PR approval. I got some feedback to make minor changes...

ramyaparimi reviewed (approved):
@varunsh-coder Thanks so much for opening a PR with this fix! I'll get this merged once tests are passing 🍏

ramyaparimi committed:
Merge branch 'main' into patch-2
